### DISCUSSION TITLE
Receiving strange error code :  "PGRST201" "Could not embed because more than one relationship

### DISCUSSION THREAD

**Discussion by hallodom** [supabase/supabase#22870] [Questions]
Hi everyone, would really appreciate any hep on this 

I have added a column to a table that referenced a foreign key that was already use on the same table. 

This key represents user ID or 'profiles' in my database design. 

The table stores a history of images uploaded, the ID of the user who uploaded the image and then the ID of the user who last edited the image name (edited_by) column.  

It is the last edited column that seems to break the table. I added a last edited column to reference user ID as a foreign key and then began to get this error on the page where I tried to load the image log from the table:

```
Code "PGRST201"
"Try changing 'profiles' to one of the following: 'profiles!public_screenshots_edited_by_fkey', 'profiles!screenshots_profile_id_fkey'. Find the desired relationship in the 'details' key."
"Could not embed because more than one relationship was found for 'screenshots' and 'profiles'"
```

Browser console screenshot of error log:

<img width="1358" alt="Screenshot 2024-04-17 at 22 11 18" src="https://github.com/Retake-App/retake-browser-extension/assets/359858/fddb6a6c-9711-4a9e-a658-7e244c3cbeaa">

I have since deleted the 'edited_by' column but I am still seeing this error.

The user is referred to as ‘profile’ on the ‘screenshots’ table. 

Can anyone help with this? 

I found an old github thread on it but I could not understand as I'm not advanced postgres user.
https://github.com/orgs/supabase/discussions/6274

Here is my current table Scheme

```
create table
  public.screenshots (
    id uuid not null default gen_random_uuid (),
    page jsonb null,
    image jsonb null,
    coordinates jsonb null,
    dimensions jsonb null,
    screen jsonb null,
    device_details jsonb null,
    meta jsonb null,
    steps text[] null,
    profile_id uuid not null,
    org_id text not null,
    group_id uuid null,
    created_at timestamp with time zone null default now(),
    edited_at timestamp with time zone null,
    constraint screenshots_dev_pkey primary key (id),
    constraint screenshots_group_id_fkey foreign key (group_id) references groups (id) on update cascade on delete cascade,
    constraint screenshots_org_id_fkey foreign key (org_id) references orgs (id) on update cascade on delete cascade,
    constraint screenshots_profile_id_fkey foreign key (profile_id) references profiles (id) on update cascade on delete cascade
  ) tablespace pg_default;

```

**Comment by GaryAustin1** [comment]
Basically because you had two possible relations to the table profiles, the message is showing you what to put to clarify which one you want  for each case.  The ! is a hint to PostgREST on what key to use to get to profiles.

It is possible when you deleted one key the cache PostgREST uses did not pick it up.  You sometimes need to run `NOTIFY pgrst, 'reload schema'` in the SQL editor to reset after changing a foreign key. (I would not expect it to help here as you deleted the actual column though).

But you don't show your query and that is where the error is occurring.  Also you image link errors for me.
